### DISCUSSION TITLE
`todo msg` vs `todo as msg`

### DISCUSSION THREAD

**Discussion by sobolevn** [gleam-lang/gleam#3475] [Ideas & suggestions]
Currently you can write:

```gleam
fn testfn() {
  todo "msg"
}

pub fn main() {
  testfn()
}
```

It will produce: `Error: This has not yet been implemented`

And you can write:

```gleam
fn testfn() {
  todo as "msg"
}

pub fn main() {
  testfn()
}
```

Which will produce `Error: msg` as expected.
It parses as:

```rust
[
    Expression(
        Todo {
            kind: Keyword,
            location: SrcSpan {
                start: 1,
                end: 5,
            },
            message: None,
        },
    ),
    Expression(
        String {
            location: SrcSpan {
                start: 6,
                end: 12,
            },
            value: "text",
        },
    ),
]
```

So, just two expressions: one after another.

The same happens with `panic`.

It looks like this might be a common place for errors.
That's why I propose to discuss: 
- Is it really a common place for errors? I made this error myself. Was very surprised to see that the error message was different
- Is it worth it to change this to be a parse error? this is a breaking change on paper, but is it really in real life? Are there any real usages of `todo` near the next meaningful expression?

If we decide to make this an error, I have a PR ready :)

**Comment by lpil** [comment]
Gleam is not a whitespace sensitive language so `todo msg` is a `todo` followed by the variable `msg`. If you run the formatter it'll be formatted like so:

```gleam
fn testfn() {
  todo
  "msg"
}
```

**Reply by sobolevn** [reply]
Yes :)
I understand that, that's why I attached the generated AST for this case.
I am asking if this code is actually useful for anything productive? 

Code after `todo` should not be executed, so in this case `"msg"` will be unreachable and useless. Are there any other cases where `todo "msg"` might do something in a real-life use-case?

**Reply by giacomocavalieri** [reply]
The way I see it, Todo is for rapidly prototyping, adding warnings for unreachable code after a todo wouldn't be helpful since we already emit a warning for the todo itself. Rust does it and I always found it a bit annoying because I just get double the warnings for no real gain.

I think it's really common to have a todo followed by some valid expression, I do it all the time when prototyping things :)

**Reply by inoas** [reply]
I like `todo as string.inspect(my_var)` (or pprint.inspect)
